drivers/tee/optee/rpc.c

```diff
@@ -310,6 +312,45 @@ static void handle_rpc_func_cmd_shm_free(struct tee_context *ctx,
 	arg->ret = TEEC_SUCCESS;
 }
 
+static void handle_rpc_func_cmd_bench_reg_buffer(struct optee_msg_arg *arg)
+{
+	u64 size;
+	u64 type;
+
+	if (arg->num_params != 1)
+		goto bad;
+
+	if ((arg->params[0].attr & OPTEE_MSG_ATTR_TYPE_MASK) !=
+			OPTEE_MSG_ATTR_TYPE_VALUE_INPUT)
+		goto bad;
+
+	type = arg->params[0].u.value.a;
+	switch (type) {
+	case OPTEE_MSG_RPC_CMD_BENCH_REG_NEW:
+		size = arg->params[0].u.value.c;
+		optee_bench_ts_global =
+			memremap(arg->params[0].u.value.b, size, MEMREMAP_WB);
+
+		if (!optee_bench_ts_global)
+			goto bad;
+
+		break;
+	case OPTEE_MSG_RPC_CMD_BENCH_REG_DEL:
+		if (optee_bench_ts_global)
+			memunmap(optee_bench_ts_global);
+
+		optee_bench_ts_global = NULL;
+			break;
```

Comment:
If it's not the GitHub gui that fools me, then this break should move back one tab.